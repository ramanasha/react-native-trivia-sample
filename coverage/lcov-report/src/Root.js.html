<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/Root.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\prettify.css" />
    <link rel="stylesheet" href="..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\index.html">All files</a> / <a href="index.html">src</a> Root.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/45</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/20</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/15</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34</td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js"><span class="cstat-no" title="statement not covered" >/** @flow */<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span>
&nbsp;
import Reac<span class="cstat-no" title="statement not covered" >t from 'react';<span class="cstat-no" title="statement not covered" ></span></span>
import { St<span class="cstat-no" title="statement not covered" >oreProvider } from 's</span>rc/redux';
import { Root as<span class="cstat-no" title="statement not covered" > RootContainer, StylePr</span>ovider } from 'native-base';
import { Spla<span class="cstat-no" title="statement not covered" >sh } from 'src/screens'</span>;
import { Provide<span class="cstat-no" title="statement not covered" >r } from 'react-redux';</span>
import { Connect<span class="cstat-no" title="statement not covered" >edNavigator } from 'src/na</span>vigation';
import { The<span class="cstat-no" title="statement not covered" >me } from 'src/styles'</span>;
import { Per<span class="cstat-no" title="statement not covered" >sistGate } from 'redux-persist/src/integration/</span>react';
import config<span class="cstat-no" title="statement not covered" > from '../config/default.json';<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ></span></span></span></span>
&nbsp;
<span class="cstat-no" title="statement not covered" >const [store, persistor] = StoreProvider.init</span>(config);<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ></span></span></span>
&nbsp;
const Roo<span class="cstat-no" title="statement not covered" >t = funct<span class="fstat-no" title="function not covered" >ion (</span>): React$Element&lt;Provider&gt; {</span>
<span class="cstat-no" title="statement not covered" >    return (</span>
        &lt;Provider store={store}&gt;
            &lt;StyleProvider style={Theme.getTheme(Theme.variables)}&gt;
                &lt;PersistGate
                    loading={&lt;Splash /&gt;}
                    persistor={persistor}
                    on<span class="fstat-no" title="function not covered" >BeforeLift={(</span>) <span class="cstat-no" title="statement not covered" >=&gt; StoreProvider.serverSync(store)}</span>
                &gt;
                    &lt;RootContainer&gt;
                        &lt;ConnectedNavigator /&gt;
                    &lt;/RootContainer&gt;
                &lt;/PersistGate&gt;
            &lt;/StyleProvider&gt;
        &lt;/Provider&gt;
    );
};<span class="cstat-no" title="statement not covered" ></span>
&nbsp;
export default Root;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Dec 03 2017 13:47:16 GMT-0500 (SA Pacific Standard Time)
</div>
</div>
<script src="..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\sorter.js"></script>
</body>
</html>
