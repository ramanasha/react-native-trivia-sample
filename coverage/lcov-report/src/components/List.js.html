<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/List.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">src/components</a> List.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">40% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/10</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">// @flow
&nbsp;
import React from 'react';
import { FlatList, SectionList } from 'react-native';
&nbsp;
class List extends React.PureComponent&lt;{
    list: Array&lt;any&gt;,
    renderItem: () =&gt; mixed,
    renderSectionHeader?: () =&gt; mixed,
    scrollToEnd?: boolean,
    idField?: string}&gt; {
    _list: FlatList | SectionList;
&nbsp;
    static defaultProps = {
        renderSectionHeader: null,
        scrollToEnd: false,
        idField: 'id',
    };
&nbsp;
    componentDidMount() {
        const { scrollToEnd } = this.props;
        <span class="missing-if-branch" title="if path not taken" >I</span>if (scrollToEnd) {
<span class="cstat-no" title="statement not covered" >            this._list.scrollToEnd({ animated: true });</span>
        }
    }
&nbsp;
    render(): React$Element&lt;FlatList | SectionList&gt; {
        const { list, renderItem, renderSectionHeader, idField } = this.props;
&nbsp;
        <span class="missing-if-branch" title="if path not taken" >I</span>if (renderSectionHeader) {
<span class="cstat-no" title="statement not covered" >            return (</span>
                &lt;SectionList
                    sections={list}
                    keyExtractor={<span class="fstat-no" title="function not covered" >(i</span>tem) =&gt; <span class="cstat-no" title="statement not covered" >item[idField]}</span>
                    renderItem={renderItem}
                    renderSectionHeader={renderSectionHeader}
                /&gt;
            );
        }
&nbsp;
        return (
            &lt;FlatList
                data={list}
                keyExtractor={<span class="fstat-no" title="function not covered" >(i</span>tem) =&gt; <span class="cstat-no" title="statement not covered" >item[idField]}</span>
                renderItem={renderItem}
                ref={<span class="fstat-no" title="function not covered" >(c</span>) =&gt; (<span class="cstat-no" title="statement not covered" >this._list = c)</span>}
            /&gt;
        );
    }
}
&nbsp;
export default List;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Dec 03 2017 13:47:16 GMT-0500 (SA Pacific Standard Time)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
