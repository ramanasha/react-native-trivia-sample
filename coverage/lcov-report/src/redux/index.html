<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/redux</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> src/redux
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">78.67% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>59/75</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">76.67% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>23/30</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>16/20</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">77.46% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>55/71</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="actions.js"><a href="actions.js.html">actions.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="11" class="abs high">11/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="11" class="abs high">11/11</td>
	</tr>

<tr>
	<td class="file high" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	</tr>

<tr>
	<td class="file high" data-value="reducers.js"><a href="reducers.js.html">reducers.js</a></td>
	<td data-value="97.37" class="pic high"><div class="chart"><div class="cover-fill" style="width: 97%;"></div><div class="cover-empty" style="width:3%;"></div></div></td>
	<td data-value="97.37" class="pct high">97.37%</td>
	<td data-value="38" class="abs high">37/38</td>
	<td data-value="95" class="pct high">95%</td>
	<td data-value="20" class="abs high">19/20</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="97.3" class="pct high">97.3%</td>
	<td data-value="37" class="abs high">36/37</td>
	</tr>

<tr>
	<td class="file high" data-value="selectors.js"><a href="selectors.js.html">selectors.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	</tr>

<tr>
	<td class="file high" data-value="state.js"><a href="state.js.html">state.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

<tr>
	<td class="file low" data-value="store.js"><a href="store.js.html">store.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="15" class="abs low">0/15</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="15" class="abs low">0/15</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Dec 03 2017 13:47:16 GMT-0500 (SA Pacific Standard Time)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
