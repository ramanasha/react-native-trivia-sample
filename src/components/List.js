// @flow

import React from 'react';
import { FlatList, SectionList } from 'react-native';

class List extends React.PureComponent<{
    list: Array<any>,
    renderItem: () => mixed,
    renderSectionHeader?: () => mixed,
    scrollToEnd?: boolean,
    idField?: string}> {
    _list: FlatList | SectionList;

    static defaultProps = {
        renderSectionHeader: null,
        scrollToEnd: false,
        idField: 'id',
    };

    componentDidMount() {
        const { scrollToEnd } = this.props;
        if (scrollToEnd) {
            this._list.scrollToEnd({ animated: true });
        }
    }

    render(): React$Element<FlatList | SectionList> {
        const { list, renderItem, renderSectionHeader, idField } = this.props;

        if (renderSectionHeader) {
            return (
                <SectionList
                    sections={list}
                    keyExtractor={(item) => item[idField]}
                    renderItem={renderItem}
                    renderSectionHeader={renderSectionHeader}
                />
            );
        }

        return (
            <FlatList
                data={list}
                keyExtractor={(item) => item[idField]}
                renderItem={renderItem}
                ref={(c) => (this._list = c)}
            />
        );
    }
}

export default List;
