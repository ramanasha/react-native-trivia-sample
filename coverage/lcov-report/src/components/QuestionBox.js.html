<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/QuestionBox.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">src/components</a> QuestionBox.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/3</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">// @flow
&nbsp;
import React from 'react';
import { Dimensions } from 'react-native';
import { Card, CardItem } from 'native-base';
import { AutoHeightWebView } from 'src/components';
import { Styles, Colors } from 'src/styles';
import type { Question } from 'src/redux/state';
&nbsp;
class QuestionBox extends React.PureComponent&lt;{ question: Question }&gt; {
    render(): React$Element&lt;Card&gt; {
        const { question } = this.props;
        const minHeight = 100;
&nbsp;
        return (
            &lt;Card style={Styles.center}&gt;
                &lt;CardItem&gt;
                    &lt;AutoHeightWebView
                        source={{ html: `&lt;p style="font-size: 26px; color:${Colors.gray}"&gt;${question.question}&lt;/p&gt;` }}
                        style={{ width: Dimensions.get('window').width - 130, minHeight }}
                        enableAnimation
                    /&gt;
                &lt;/CardItem&gt;
            &lt;/Card&gt;
        );
    }
}
&nbsp;
export default QuestionBox;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Dec 03 2017 13:47:16 GMT-0500 (SA Pacific Standard Time)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
