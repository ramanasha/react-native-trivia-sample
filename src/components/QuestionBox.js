// @flow

import React from 'react';
import { Dimensions } from 'react-native';
import { Card, CardItem } from 'native-base';
import { AutoHeightWebView } from 'src/components';
import { Styles, Colors } from 'src/styles';
import type { Question } from 'src/redux/state';

class QuestionBox extends React.PureComponent<{ question: Question }> {
    render(): React$Element<Card> {
        const { question } = this.props;
        const minHeight = 100;

        return (
            <Card style={Styles.center}>
                <CardItem>
                    <AutoHeightWebView
                        source={{ html: `<p style="font-size: 26px; color:${Colors.gray}">${question.question}</p>` }}
                        style={{ width: Dimensions.get('window').width - 130, minHeight }}
                        enableAnimation
                    />
                </CardItem>
            </Card>
        );
    }
}

export default QuestionBox;
