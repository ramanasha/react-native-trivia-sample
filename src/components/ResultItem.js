// @flow

import React from 'react';
import { Dimensions } from 'react-native';
import { ListItem, Left, Body, Icon, Text } from 'native-base';
import { AutoHeightWebView } from 'src/components';
import { Styles, Colors } from 'src/styles';
import type { Result } from 'src/redux/state';

class ResultItem extends React.PureComponent<{ result: Result }> {
    render(): React$Element<ListItem> {
        const { result } = this.props;
        const minHeight = 40;

        let icon;
        if (result.answerIsCorrect) {
            icon = <Icon active name="check-circle-outline" style={[Styles.greenColor, Styles.icon, Styles.center]} />;
        } else {
            icon = <Icon active name="minus-box-outline" style={[Styles.redColor, Styles.icon, Styles.center]} />;
        }

        return (
            <ListItem thumbnail>
                <Left style={[Styles.iconContainer, Styles.top]}>
                    {icon}
                </Left>
                <Body>
                    <AutoHeightWebView
                        source={{ html: `<p style="font-size: 16px; color:${Colors.gray}">${result.question}</p>` }}
                        style={{ width: Dimensions.get('window').width - 130, minHeight }}
                        enableAnimation
                    />
                    {result.userAnswer != null &&
                    <Text note style={Styles.rightText}>your answer: {result.userAnswer.toString()}</Text>
                    }
                </Body>
            </ListItem>
        );
    }
}

export default ResultItem;
