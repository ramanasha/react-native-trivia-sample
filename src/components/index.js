// @flow

import QuestionBox from 'src/components/QuestionBox';
import ResultItem from 'src/components/ResultItem';
import List from 'src/components/List';
import AutoHeightWebView from 'react-native-autoheight-webview';

export {
    QuestionBox,
    ResultItem,
    List,
    AutoHeightWebView,
};
