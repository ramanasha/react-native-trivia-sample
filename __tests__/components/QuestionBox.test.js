// @flow

import React from 'react';
import { shallow } from 'enzyme';
import { QuestionBox } from '../../src/components';
import { questions } from '../setup';

it('renders the QuestionBox component correctly', () => {
    const wrapper = shallow(<QuestionBox question={questions[0]} />);
    expect(wrapper).toMatchSnapshot();
});
