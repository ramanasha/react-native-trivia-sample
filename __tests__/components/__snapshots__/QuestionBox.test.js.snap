// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the QuestionBox component correctly 1`] = `
<Styled(Card)
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "justifyContent": "center",
    }
  }
>
  <Styled(CardItem)>
    <AutoHeightWebView
      animationDuration={555}
      enableAnimation={true}
      heightOffset={12}
      scalesPageToFit={false}
      source={
        Object {
          "html": "<p style=\\"font-size: 26px; color:#939598\\">In &quot;The Sims&quot; series, the most members in a household you can have is 8.</p>",
        }
      }
      style={
        Object {
          "minHeight": 100,
          "width": 620,
        }
      }
    />
  </Styled(CardItem)>
</Styled(Card)>
`;
